app/controllers/api/v2/courses_controller.rb

```diff
@@ -47,19 +47,15 @@ def publish
       end
 
       def update
-        enrichment = if @course.enrichments.draft.any?
-                       @course.enrichments.draft.first
-                     else
-                       @course.enrichments.new(status: 'draft')
-                     end
-
+        enrichment = first_draft_or_new_enrichment
         enrichment.assign_attributes(update_params)
         enrichment.save
 
         site_ids = params[:course][:sites_ids]
         @course.sites = @provider.sites.where(id: site_ids) if site_ids.present?
+        @course.errors[:sites] << "^You must choose at least one location" if site_ids == []
```

Comment:
```suggestion
        @course.errors[:sites] << "^You must choose at least one location" if site_ids.blank?
```

Reply:
This won't work, because it catches `nil` too:

```ruby
[1] pry(main)> nil.blank?
=> true
[2] pry(main)> nil == []
=> false
```

I thought about doing `!site_ids.nil? && site_ids.empty?` but `== []` seems more explicit...